packages/core/src/actions/actions.ts

```diff
@@ -56,6 +56,60 @@ export const UPDATE_I18N = 'jsonforms/UPDATE_I18N' as const;
 export const ADD_DEFAULT_DATA = 'jsonforms/ADD_DEFAULT_DATA' as const;
 export const REMOVE_DEFAULT_DATA = 'jsonforms/REMOVE_DEFAULT_DATA' as const;
 
+export type UpdateArrayContext =
+  | { type: 'ADD'; values: any[] }
+  | { type: 'REMOVE'; indices: number[] }
+  | { type: 'MOVE'; moves: { from: number; to: number }[] };
+
+export const isUpdateArrayContext = (
+  context: object
+): context is UpdateArrayContext => {
+  if (!('type' in context)) {
+    return false;
+  }
+  if (typeof context.type !== 'string') {
+    return false;
+  }
+  switch (context.type) {
+    case 'ADD': {
+      if (
+        'values' in context &&
+        Array.isArray(context.values) &&
+        context.values.length > 0
+      ) {
+        const firstValueType = typeof context.values[0];
+        return context.values.every((v) => typeof v === firstValueType);
+      }
+      return false;
+    }
+    case 'REMOVE': {
+      return (
+        'indices' in context &&
+        Array.isArray(context.indices) &&
+        context.indices.length > 0 &&
+        context.indices.every((i) => typeof i === 'number')
+      );
+    }
+    case 'MOVE': {
+      return (
+        'moves' in context &&
+        Array.isArray(context.moves) &&
+        context.moves.length > 0 &&
+        context.moves.every(
+          (m) =>
+            typeof m === 'object' &&
+            'from' in m &&
```

Comment:
This will crash in case `m` is `null`. We should add a `!!m` check too.

Reply:
I will add a test case and adapt the code if necessary